connector_importer/models/recordset.py

```diff
@@ -272,24 +271,21 @@ def generate_report(self):
     def _get_importers(self):
         importers = OrderedDict()
         for importer_config in self.available_importers():
-            kwargs = {
-                "options": importer_config.options,
-            }
-            model = self.env["ir.model"]._get(importer_config.model)
-            with self.backend_id.with_context(**importer_config.context).work_on(
-                self._name, **kwargs
-            ) as work:
-                importers[model] = work.component_by_name(
-                    importer_config.importer, model_name=importer_config.model
-                )
+            model_record = self.env["ir.model"]._get(importer_config.model)
+            importers[model_record] = get_importer_for_config(
+                self.backend_id, self._name, importer_config
+            )
         return importers
 
     @api.depends("import_type_id")
     def _compute_docs_html(self):
         template = self.env.ref("connector_importer.recordset_docs")
         for item in self:
             item.docs_html = False
-            if isinstance(item.id, models.NewId):
+            if isinstance(item.id, models.NewId) or not item.backend_id:
+                # Surprise surprise: when editing a new recordset
+                # if you hit `configure source` btn
+                # the record will be saved but the backend can be null :S
```

Comment:
Not sure to understand, if you create a new recordset from the UI, as there is no `backend_id` on the form it is expected to not have it defined here no?